File: day1/day1.py
lines = []
with open("./input.txt", "r") as f:
    lines = f.readlines()

count = 0
left = []
right = []
for line in lines:
    st = line.strip()
    # print(st)
    sp = st.split()
    # print(sp)

    left.append(int(sp[0]))
    right.append(int(sp[1]))

left = sorted(left)
right = sorted(right)
for i in range(len(left)):
    l = left[i]
    r = right[i]
    count += abs(r - l)
# print(count)

map = {}

for j in range(len(left)):
    map[left[j]] = 0

print(map)

for k in range(len(right)):
    print(right[k])
    if right[k] in map:
        map[right[k]] += 1

ans = 0
for key in map:
    ans += key * map[key]
print(ans)

File: day1/day1.js
import { readFile } from '../utils/utils.js';

// const file = readFile(import.meta.dirname, 'test.txt');
const file = readFile(import.meta.dirname, 'input.txt');

const lines = file.split('\n');
const alist = [];
const blist = [];
for (let line of lines) {
    const [a,b] = line.trim().split('   ');
    if (a && b) {
        alist.push(Number(a));
        blist.push(Number(b));
    }
}

alist.sort();
blist.sort();

let part1 = 0;
for (let i = 0; i < alist.length; i++) {
    let ai = alist[i];
    let bi = blist[i];
    part1 += Math.abs(ai - bi);
}
console.log('Part 1: ', part1);

const map = {};
for (let i = 0; i < alist.length; i++) {
    let ai = alist[i];
    map[ai] = 0;
}

for (let j = 0; j < blist.length; j++) {
    let bi = blist[j];
    map[bi] += 1;
}

console.log(map);

let part2 = 0;
for (let [key, value] of Object.entries(map)) {
    if (value) {
        part2 += key * value;
    }
}
console.log('Part 2: ', part2);
